pkg/kubelet/rkt/rkt.go

```diff
@@ -1543,16 +1574,9 @@ func (r *Runtime) KillPod(pod *api.Pod, runningPod kubecontainer.Pod, gracePerio
 		return err
 	}
 
-	// Clean up networking; use running pod details since 'pod' can be nil
-	if pod == nil || !kubecontainer.IsHostNetworkPod(pod) {
-		err := r.cleanupPodNetwork(&api.Pod{
-			ObjectMeta: api.ObjectMeta{
-				UID:       runningPod.ID,
-				Name:      runningPod.Name,
-				Namespace: runningPod.Namespace,
-			},
-		})
-		if err != nil {
+	// Clean up networking; use running pod details since 'pod' can be nil.
+	if pod != nil && !kubecontainer.IsHostNetworkPod(pod) {
```

Comment:
Doens't this block ignore the comment just above?  'pod' can be nil here if a pod that's already been deleted from the API is now being cleaned up.  Shouldn't it be "if pod == nil"?  Also, cleanupPodNetworkFromServiceFile() already handles hostnetwork pods so we don't need that check.  Actually, I don't think we need any check here at all perhaps?  Just unconditionally do the cleanup.